rustls/src/crypto/signer.rs

```diff
@@ -105,6 +112,25 @@ impl CertifiedKey {
         }
     }
 
+    /// Verify the consistency of this [`CertifiedKey`]'s public and private keys.
+    /// This is done by performing a comparison of SubjectPublicKeyInfo bytes.
+    pub fn keys_match(&self) -> Result<(), Error> {
+        self.key
+            .public_key()
+            .map(|spki| {
+                self.end_entity_cert()
+                    .and_then(ParsedCertificate::try_from)
+                    .map(|ee| ee.subject_public_key_info() == spki)
+            })
+            // Error case: The signing key cannot produce a public key
+            // Error case: We've failed to parse this end-entity certificate
+            .ok_or(InconsistentKeys::Unknown)??
+            // Looking good: The SPKI values are equal
+            .then_some(())
+            // Error case: The SPKI values are not equal and these keys do not match
+            .ok_or(InconsistentKeys::KeyMismatch.into())
```

Comment:
I think chaining style is suboptimal for this case. Suggest more like this:

```rust
let key_spki = match self.key.public_key() {
    Some(key) => key,
    None => return Err(InconsistentKeys::Unknown.into()),
};

let cert = ParsedCertificate::try_from(self.end_entity_cert()?)?;
match key_spki == cert.subject_public_key_info() {
    true => Ok(())
    false => Err(InconsistentKeys::KeyMismatch.into()),
}
```

Reply:
Added nearly exactly this in [87bb093](https://github.com/rustls/rustls/pull/1954/commits/87bb09334b54d07e5f57b61ca3c9237af50849d0)! Nearly = missing a comma